src/originalBudgetTrackingApp/export/exportTransactions.ts

```diff
@@ -1,28 +1,42 @@
 import { Config } from '@/originalBudgetTrackingApp/configManager/configManager';
+import { BudgetTrackingEventEmitter, EventNames } from '@/originalBudgetTrackingApp/eventEmitters/EventEmitter';
 import { EnrichedTransaction } from '@/originalBudgetTrackingApp/commonTypes';
 import _ from 'lodash';
 import outputVendors from '@/originalBudgetTrackingApp/export/outputVendors';
 
 export async function createTransactionsInExternalVendors(
   outputVendorsConfig: Config['outputVendors'],
   companyIdToTransactions: Record<string, EnrichedTransaction[]>,
-  startDate: Date
+  startDate: Date, eventEmitter: BudgetTrackingEventEmitter
 ) {
+  await eventEmitter.emit(EventNames.EXPORT_PROCESS_START);
   const executionResult = {};
   const allTransactions = _.flatten(Object.values(companyIdToTransactions));
 
   for (let j = 0; j < outputVendors.length; j++) {
     const outputVendor = outputVendors[j];
     if (outputVendorsConfig[outputVendor.name]?.active) {
+      const baseExporterEventData = { name: outputVendor.name.toString(), allTransactions };
+
       await outputVendor.init?.(outputVendorsConfig);
-      console.log(`Start creating transactions in ${outputVendor.name}`);
-      const vendorResult = await outputVendor.exportTransactions(allTransactions, startDate, outputVendorsConfig);
-      console.log(`Finished creating transactions in ${outputVendor.name}`);
-      executionResult[outputVendor.name] = vendorResult;
+      await eventEmitter.emit(EventNames.EXPORTER_START, baseExporterEventData);
+      try {
+        const vendorResult = await outputVendor.exportTransactions({
+          transactionsToCreate: allTransactions, startDate, outputVendorsConfig, eventEmitter
+        });
+        await eventEmitter.emit(EventNames.EXPORTER_END, baseExporterEventData);
+        executionResult[outputVendor.name] = vendorResult;
+      } catch (e) {
+        await eventEmitter.emit(EventNames.EXPORTER_ERROR, { ...baseExporterEventData, error: e });
+        throw e;
```

Comment:
Need to think/declare the error handling with the `eventEmitter`.
Since the error handling work with throwing up, it might lead to `emit` in each throwing level. In that case we can declare that the error events are just to emit an **info** and shouldn't affect the logic because we actually throw.

Another option is to throw custom error inside the code, so you can throw the `eventName` through all the levels and both `emit` and `throw` it out once, in the highest level, but with the specific eventName.

Reply:
I was thinking about error handling as well. There are cases where we would want it to stop the flow and cases where we won't. For example, if one importer is failing, it is not necessarily a reason to crash the whole process. I would expect the other ones to complete successfully  and the one that fails to fail (which will be reflected in the ui of the progress once that exists).

So, in this case I would remove the throw from here all together and just emit the event (not doing this yet because since we don't have a status indication in the ui at the moment, if an errors isn't thrown it will look like it succeeded even though it failed).

What still looks like it doesn't make sense is that ynab emits and throws, and then the exporter emits and throws again.
The reason that code is there is because when I didn't have a catch around the ynab code, I didn't get the full details of the error that was thrown. This is probably the issue node has with async stack traces.

Reply:
I think I understand. Not talking about specific implementation, in general we need to handle errors for each exporter and importer, and in high level I think we need to `emit` that error and **collect** it (for such `results` object we already have, I think).

If we encapsulate this specific case, we still have other optional errors, but I can guess it is more simple and we can only emit for these cases in the highest level.

Ok, it is not important how to implement this, it is just important to know about the potential issue and how we can possibly solve it.

> What still looks like it doesn't make sense is that ynab emits and throws, and then the exporter emits and throws again.
> The reason that code is there is because when I didn't have a catch around the ynab code, I didn't get the full details of the error that was thrown. This is probably the issue node has with async stack traces.

This is something I didn't understand, maybe because I don't familiar with the `ynab` code.

I think we need to open an issue about error handling and emitting, first to work on it, of course, but also to document our ideas and concern.

Reply:
I created an issue for consistent error handling #105